modules/lofreq/1.0/lofreq.smk

```diff
@@ -0,0 +1,116 @@
+#!/usr/bin/env snakemake
+
+
+##### ATTRIBUTION #####
+
+
+# Original Author:  Bruno Grande
+# Module Author:    Bruno Grande
+# Contributors:     N/A
+
+
+##### SETUP #####
+
+
+# Import package with useful functions for developing analysis modules
+import oncopipe as op
+
+# Setup module and store module-specific configuration in `CFG`
+# `CFG` is a shortcut to `config["lcr-modules"]["lofreq"]`
+CFG = op.setup_module(
+    name = "lofreq",
+    version = "1.0",
+    subdirectories = ["inputs", "lofreq", "outputs"],
+)
+
+# Define rules to be run locally when using a compute cluster
+localrules:
+    _lofreq_input_bam,
+    _lofreq_output_vcf,
+    _lofreq_all,
+
+
+##### RULES #####
+
+
+# Symlinks the input files into the module results directory (under '00-inputs/')
+rule _lofreq_input_bam:
+    input:
+        bam = CFG["inputs"]["sample_bam"],
+        bai = CFG["inputs"]["sample_bai"]
+    output:
+        bam = CFG["dirs"]["inputs"] + "bam/{seq_type}--{genome_build}/{sample_id}.bam",
+        bai = CFG["dirs"]["inputs"] + "bam/{seq_type}--{genome_build}/{sample_id}.bam.bai"
+    run:
+        op.relative_symlink(input.bam, output.bam)
+        op.relative_symlink(input.bai, output.bai)
+
+
+# Run LoFreq in somatic variant calling mode
+rule _lofreq_run:
+    input:
+        tumour_bam = CFG["dirs"]["inputs"] + "bam/{seq_type}--{genome_build}/{tumour_id}.bam",
```

Comment:
For the Strelka module we decided that we would always merge the snv and indel vcf files and delete the individual variant type files. Could you do the same here? 

Do the VCF files need any filtering? And are there any intermediate files output by lofreq or is it just the vcf files?

Reply:
In regards to filtering, I found that there is a default filter in lofreq in and it says `Predicted variants are already filtered using default parameters (which include coverage, strand-bias, SNV-quality etc). There will usually be no need for you to filter the output again`. I looked further on what are those default parameters, and see they are mentioned here `the main filtering step is working on the variant qualities (which are converted p-values) and it's by default based on Bonferroni correction and a significance threshold of 0.01`. Examining the output, I do see for several files that I checked that in the output file, there are already only snvs with PASS in the filter column. So, I think that the filtering for these vcfs is not needed.
For the intermediate files, there are no temp files created. Regarding output, it produces total 26 files (including logs and indices), of which we are really interested in 4 (those are that got symlinked into the 99-outputs). Are you thinking about clearing those files? I think I would keep them, since it includes the calls for both normal and tumour samples.

Reply:
Do you have an example loFreq output directory I could have a look at?

Reply:
I suspect everything else here that I haven't indicated with a comment should be wrapped in temp() so it is deleted after.

```
normal_relaxed.log                  
normal_stringent.snvs.vcf.gz.tbi             
somatic_raw.snvs.vcf.gz.tbi        
tumor_stringent.snvs.vcf.gz
normal_relaxed.vcf.gz               
somatic_final.indels.vcf.gz                  #merge and output as full mutation VCF
somatic_final.snvs.vcf.gz                  #merge and output as full mutation VCF, regenerate index
tumor_relaxed.log                  
tumor_stringent.snvs.vcf.gz.tbi
normal_relaxed.vcf.gz.tbi           
somatic_final.indels.vcf.gz.tbi              
somatic_final.snvs.vcf.gz.tbi              
tumor_relaxed.vcf.gz
normal_stringent.indels.vcf.gz      
somatic_raw.indels.vcf.gz                  
tumor_relaxed.vcf.gz.tbi
normal_stringent.indels.vcf.gz.tbi  
somatic_final_minus-dbsnp.indels.vcf.gz.tbi  
somatic_final_minus-dbsnp.indels.vcf.gz      #merge and output as snp-filtered vcf
somatic_final_minus-dbsnp.snvs.vcf.gz.tbi  
somatic_final_minus-dbsnp.snvs.vcf.gz      #merge and output as snp-filtered vcf, regenerate index
somatic_raw.indels.vcf.gz.tbi              
tumor_stringent.indels.vcf.gz
normal_stringent.snvs.vcf.gz        
somatic_raw.snvs.vcf.gz                    
tumor_stringent.indels.vcf.gz.tbi
```

Reply:
Kostia and I discussed that since most of these files are very small we could keep them for the sake of understanding the calls loFreq makes, especially if we discover things are quite different from Strelka or Mutect.

Reply:
Fair enough. I suggest at the very least we delete normal_relaxed.vcf.gz